src/scripts/main.js

```diff
@@ -1,3 +1,320 @@
 'use strict';
 
-// write code here
+const tableHead = document.querySelector('thead');
+const tableBody = document.querySelector('tbody');
+const tableRows = tableBody.rows;
+let toggle = true;
+let valueForSort = 0;
+let special;
+
+function setSalaryValue(num) {
+  return `$${num.toLocaleString('en-US')}`;
+};
+
+function getSalaryValue(str) {
+  return +str.replace(/[,$]/g, '');
+};
+
+tableHead.addEventListener('click', e => {
+  const index = e.target.cellIndex;
+  const headTitle = e.target.innerHTML;
+
+  const sortRows = [...tableRows].sort((a, b) => {
+    let prev = a.cells[index].innerHTML;
+    let next = b.cells[index].innerHTML;
+
+    if (headTitle === 'Salary' || headTitle === 'Age') {
+      prev = getSalaryValue(prev);
+      next = getSalaryValue(next);
+
+      return prev - next;
+    }
+
+    return prev.localeCompare(next);
+  });
+
+  if (!toggle) {
+    sortRows.reverse();
+  }
+
+  tableBody.append(...sortRows);
+
+  if (valueForSort !== index) {
+    toggle = !toggle;
+  }
```

Comment:
You can simplify the toggling of the 'toggle' variable by using a single line: `toggle = (valueForSort !== index) ? !toggle : toggle;`.